### PR TITLE
Add resnet inference jupyter notebook.

### PR DESCRIPTION
**Pull request by silvasean**
This takes the example from torchscript_resnet18_e2e.py and puts it into
a slightly cleaned up notebook form.

It's still a little rough around the edges. Areas for improvement:
- Installation / setup.
- API usability.

Also,
- Add `npcomp-backend-to-iree-frontend-pipeline` since we will be adding
  more stuff there.
- Slight cleanups.

### CODE REVIEW

**Review by cathyzhyi — APPROVED** [review]
LGTM

File: lib/Backend/IREE/LowerLinkage.cpp [hunk 1]
```diff
@@ -16,10 +16,20 @@ using namespace mlir::NPCOMP;
...
-#include "npcomp/Backend/IREE/Passes.h.inc"
-} // end namespace
+// This pass lowers the public ABI of the module to the primitives exposed by
+// the refbackrt dialect.
+class LowerLinkagePass : public LowerLinkageBase<LowerLinkagePass> {
+  void runOnOperation() override {
+    ModuleOp module = getOperation();
+    for (auto func : module.getOps<FuncOp>()) {
+      if (func.getVisibility() == SymbolTable::Visibility::Public)
+        func->setAttr("iree.module.export", UnitAttr::get(&getContext()));
```
Comment (stellaraccident) [inline lib/Backend/IREE/LowerLinkage.cpp:26]: I think we don't need this anymore

Comment (silvasean) [reply lib/Backend/IREE/LowerLinkage.cpp:26]: Awesome. :)

Comment (silvasean) [reply lib/Backend/IREE/LowerLinkage.cpp:26]: Removed in https://github.com/llvm/mlir-npcomp/pull/273

**Pull request merged**